### PR TITLE
Route conversion for the List and Get routes for security groups

### PR DESCRIPTION
**Pull request by btassone**
Removes the old ProxyRequestHandler calls to the ruby api on security groups for the list and get routes. These routes have been updated to use functions inside the go api instead.

**handlers_security.go**
* Added SecurityGroupListHandler function.
* Added SecurityGroupGetHandler function.

**routes.go**
* Removed reference to the ProxyRequestHandler function for the security groups List route and replaced it with the SecurityGroupListHandler function.
* Removed reference to the ProxyRequestHandler function for the security groups Get route and replace it with the SecurityGroupGetHandler function.

**security.go**
* Added ListSecurityGroups function
* Added the GetSecurityGroup function

**types.go**
* Added toEc2SecurityGroupResponse function
* Added toEc2SecurityGroupIpPermissions function
* Added Ec2SecurityGroupResponse type
* Added Ec2SecurityGroupIpPermission type
* Added Ec2SecurityGroupIpRange type
* Added Ec2SecurityGroupIpv6Range type
* Added Ec2SecurityGroupPrefixListId type
* Added Ec2SecurityGroupUserIdGroupPair type

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ k8s/k8s-config.yaml
 vendor
 .vscode
 debug
+.idea/
```
Comment (fishnix) [inline .gitignore:7]: 👀 newline

File: api/handlers_security.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
+package api
```
Comment (fishnix) [inline api/handlers_security.go:1]: not a big deal, but let's name the file `handlers_security_groups.go` or `handlers_sgs.go`

File: ec2/security.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+func (e *Ec2) ListSecurityGroups(ctx context.Context, org string) ([]map[string]*string, error) {
+	log.Infof("listing ec2 security groups (org: '%s')", org)
+
+	var filters []*ec2.Filter
+	if org != "" {
+		filters = []*ec2.Filter{inOrg(org)}
+	}
+
+	input := ec2.DescribeSecurityGroupsInput{
+		GroupIds: aws.StringSlice([]string{}),
```
Comment (fishnix) [inline ec2/security.go:21]: ✂️ you dont need this field

File: ec2/security.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		Filters:  filters,
+	}
+
+	out, err := e.Service.DescribeSecurityGroupsWithContext(ctx, &input)
+	if err != nil {
+		return nil, ErrCode("listing security groups", err)
+	}
+
+	log.Debugf("returning list of %d snapshots", len(out.SecurityGroups))
```
Comment (fishnix) [inline ec2/security.go:30]: `security groups`

File: ec2/security.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		}
+
+		// If sgName is nil, use the GroupName on the security group as a fallback
+		if sgName == nil {
+			sgName = s.GroupName
+		}
+
+		list[i] = map[string]*string{
+			*s.GroupId: sgName,
+		}
```
Comment (fishnix) [inline ec2/security.go:51]: I would suggest a couple of things here:
- set the default value for `sgName` first and then just override it
- don't bother with the `tags` variable, its unecessary
- use the aws sdk helper utilities
- break out of the loop once you find the Name tag

something like this:

```golang
	for i, s := range out.SecurityGroups {
		sgName := s.GroupName
		for _, t := range s.Tags {
			if aws.StringValue(t.Key) == "Name" {
				sgName = t.Value
				break
			}
		}

		list[i] = map[string]*string{
			aws.StringValue(s.GroupId): sgName,
		}
	}
```

**Review by fishnix — CHANGES_REQUESTED** [review]
Looks good!  Nice work, just a few minor things.

I also see some differences, especially around empty values.  You can probably implement custom MarshalJSON  functions for those types (you can see an example for `Ec2ImageVolumeMap` in the same file.

```
➜  tmp jd  -set rb.json go.json
@ ["incoming_rules",["set"],{}]
- {"ip_protocol":"-1","ip_ranges":[],"ipv_6_ranges":[],"prefix_list_ids":[],"user_id_group_pairs":[{"description":"All servers in this space can talk to each other","group_id":"sg-abcdefghijklmo","user_id":"xxxxxxxxxxxx"}]}
+ {"from_port":0,"ip_protocol":"-1","ip_ranges":[],"ipv_6_ranges":[],"prefix_list_ids":[],"to_port":0,"user_id_group_pairs":[{"description":"All servers in this space can talk to each other","group_id":"sg-abcdefghijklmo","group_name":"","peering_status":"","user_id":"xxxxxxxxxxxx","vpc_id":"","vpc_peering_connection":""}]}
@ ["outgoing_rules",["set"],{}]
- {"ip_protocol":"-1","ip_ranges":[{"cidr_ip":"0.0.0.0/0"}],"ipv_6_ranges":[],"prefix_list_ids":[],"user_id_group_pairs":[]}
+ {"from_port":0,"ip_protocol":"-1","ip_ranges":[{"cidr_ip":"0.0.0.0/0","description":""}],"ipv_6_ranges":[],"prefix_list_ids":[],"to_port":0,"user_id_group_pairs":[]}
@ ["tags"]
- []
+ null
@ ["owner_id"]
+ "xxxxxxxxxxxx"
```

File: ec2/sgs.go [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+// ListSecurityGroups List all security groups in an aws account
+func (e *Ec2) ListSecurityGroups(ctx context.Context, org string) ([]map[string]*string, error) {
+	log.Infof("listing ec2 security groups (org: '%s')", org)
+
+	var filters []*ec2.Filter
+	if org != "" {
+		filters = []*ec2.Filter{inOrg(org)}
+	}
+
+	input := ec2.DescribeSecurityGroupsInput{
```
Comment (tenyo) [inline ec2/sgs.go:20]: I don't think you need this `input` variable here, it can just go inline with the call below

File: ec2/sgs.go [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+// GetSecurityGroup Get the given security groups by a list of ids
+func (e *Ec2) GetSecurityGroup(ctx context.Context, ids ...string) ([]*ec2.SecurityGroup, error) {
+	if len(ids) == 0 {
+		return nil, apierror.New(apierror.ErrBadRequest, "invalid input", nil)
+	}
+
+	log.Infof("getting details about security group ids %+v", ids)
+
+	input := ec2.DescribeSecurityGroupsInput{
```
Comment (tenyo) [inline ec2/sgs.go:60]: same here

**Review by tenyo — APPROVED** [review]
Looks very good 👍 Just the one comment

**Comment by fishnix** [comment]
This looks good - lets cut a pre-release and deploy it to test.  It should "just work"

**Review by fishnix — APPROVED** [review]
looks good 👍

**Pull request merged**